napari/_qt/widgets/qt_keyboard_settings.py

```diff
@@ -530,21 +571,19 @@ def value(self):
         value: dict
             Dictionary of action names and shortcuts assigned to them.
         """
-
-        value = {}
-
-        for action_name in action_manager._actions:
-            shortcuts = action_manager._shortcuts.get(action_name, [])
-            value[action_name] = list(shortcuts)
-
-        return value
+        return get_settings().shortcuts.dict(exclude_defaults=True)[
+            'shortcuts'
+        ]
```

Comment:
Just in case, not totally sure if it's expected but while checking this locally, seems like setting an alternative keybinding and then removing it can raise an error:

* Triggering the error (setting an alternative keybinding for the toggle current viewer theme action):

<details>

![alt_shortcuts](https://github.com/napari/napari/assets/16781833/2402ece5-280e-41c4-9bc9-26147070e96c)


</details>

* Traceback:

<details>

```python-traceback
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
E:\Acer\Documentos\Quansight\Napari\napari otros\kne42\napari\napari\_vendor\qt_json_builder\qt_jsonschema_form\signal.py in emit(self=<napari._vendor.qt_json_builder.qt_jsonschema_form.signal.BoundSignal object>, *args=([],))
     20     def emit(self, *args):
     21         for sub in self._subscribers:
---> 22             sub(*args)
        sub = functools.partial(<bound method ObjectSchemaWidgetMinix.widget_on_changed of <napari._vendor.qt_json_builder.qt_jsonschema_form.widgets.ObjectSchemaWidget object at 0x000001C90A8CEB90>>, 'shortcuts')
        args = ([],)
     23
     24     def connect(self, listener):

E:\Acer\Documentos\Quansight\Napari\napari otros\kne42\napari\napari\_vendor\qt_json_builder\qt_jsonschema_form\widgets.py in widget_on_changed(self=<napari._vendor.qt_json_builder.qt_jsonschema_form.widgets.ObjectSchemaWidget object>, name='shortcuts', value=[])
    672
    673     def widget_on_changed(self, name: str, value):
--> 674         self.state[name] = value
        self.state = undefined
        name = 'shortcuts'
        value = []
    675         self.on_changed.emit(self.state)
    676

E:\Acer\Documentos\Quansight\Napari\napari otros\kne42\napari\napari\_vendor\qt_json_builder\qt_jsonschema_form\widgets.py in state(self=<napari._vendor.qt_json_builder.qt_jsonschema_form.widgets.ObjectSchemaWidget object>)
    660     @state_property
    661     def state(self) -> dict:
--> 662         return {k: w.state for k, w in self.widgets.items()}
        global k = undefined
        global w.state = undefined
        global w = undefined
        self.widgets.items = <built-in method items of dict object at 0x000001C90A8D8540>
    663
    664     @state.setter

E:\Acer\Documentos\Quansight\Napari\napari otros\kne42\napari\napari\_vendor\qt_json_builder\qt_jsonschema_form\widgets.py in <dictcomp>(.0=<dict_itemiterator object>)
    660     @state_property
    661     def state(self) -> dict:
--> 662         return {k: w.state for k, w in self.widgets.items()}
        k = 'shortcuts'
        w.state = undefined
        w = <napari._vendor.qt_json_builder.qt_jsonschema_form.widgets.ShortcutsWidget object at 0x000001C90A8CF010>
        global self.widgets.items = undefined
    663
    664     @state.setter

E:\Acer\Documentos\Quansight\Napari\napari otros\kne42\napari\napari\_vendor\qt_json_builder\qt_jsonschema_form\widgets.py in state(self=<napari._vendor.qt_json_builder.qt_jsonschema_form.widgets.ShortcutsWidget object>)
    609     @state_property
    610     def state(self) -> list:
--> 611         return self.value()
        self.value = <bound method ShortcutEditor.value of <napari._vendor.qt_json_builder.qt_jsonschema_form.widgets.ShortcutsWidget object at 0x000001C90A8CF010>>
    612
    613     def setDescription(self, description: str):

E:\Acer\Documentos\Quansight\Napari\napari otros\kne42\napari\napari\_qt\widgets\qt_keyboard_settings.py in value(self=<napari._vendor.qt_json_builder.qt_jsonschema_form.widgets.ShortcutsWidget object>)
    572             Dictionary of action names and shortcuts assigned to them.
    573         """
--> 574         return get_settings().shortcuts.dict(exclude_defaults=True)[
        global get_settings.shortcuts.dict = undefined
        global exclude_defaults = undefined
    575             'shortcuts'
    576         ]

KeyError: 'shortcuts'
```

</details>

Reply:
thanks for the callout! ill look into this!

Reply:
Checked this a little bit more, so leaving a possible way to define the method that could make things work:

```python
    def value(self):
        """
        Return the actions and non-defaults shortcuts currently assigned.

        Returns
        -------
        value: list
            List of action names and shortcuts assigned to them.
        """
        shortcuts_dict = get_settings().shortcuts.dict(exclude_defaults=True)

        return shortcuts_dict.get('shortcuts', [])
```

Reply:
I think I fixed it? Could you test this again please

Reply:
Yep, I'm unable to reproduce this with the latest changes 🎉